### ISSUE TITLE
Boolean values not parsed properly to integer fields for MySQL

### ISSUE THREAD

**Issue by nicolaspfernandes** [sequelize/sequelize#10633]
## What are you doing?
```js
isPublic: {type: DataTypes.BOOLEAN, allowNull: false, defaultValue: false}

// ---------
// Later on...

Model.create({key1: 'value1', key2: 'value2'})
//or...
Model.create({key1: 'value1', key2: 'value2', isPublic: true})
```

## What do you expect to happen?
_Record being created into the database with a proper converted default value to __0 or 1__, since MySQL uses `TINYINT(1)` as the column type for boolean fields._

## What is actually happening?
`SequelizeDatabaseError: Incorrect integer value: 'false' for column 'isPublic' at row 1`
`SequelizeDatabaseError: Incorrect integer value: 'true' for column 'isPublic' at row 1`

__Dialect:__ mysql
__Dialect version:__ 2.6.0
__Database version:__ 8.0.15
__Sequelize version:__ 5.2.1
__Tested with latest release:__ Yes

**Comment by gabegorelick** [comment]
See https://github.com/sequelize/sequelize/issues/10600

**Comment by nicolasgramlich** [comment]
Note if this helps anyone. I was still getting that error after upgrading from from `4.42.0` to `5.6.0`. It resolved itself after also upgrading `mysql2` from `1.4.2` to `1.6.5` and a fresh node modules nuke+install.

**Comment by jabowery** [comment]
> Note if this helps anyone. I was still getting that error after upgrading from from `4.42.0` to `5.6.0`. It resolved itself after also upgrading `mysql2` from `1.4.2` to `1.6.5` and a fresh node modules nuke+install.

I'm still getting:

` SequelizeDatabaseError: Incorrect integer value: 'false' for column `vocascan`.`roles`.`adminRights` at row 1
`
(trying to get vocascan running as a server) despite having done:

```
$ npm install sequelize@v6.6.5
$ npm install mysql2@v2.3.0
```

Perhaps that's because I don't have a clue as to what you meant by "fresh node modules nuke+install".

**Comment by rajathongal** [comment]
This issue still persists

**Comment by WikiRik** [comment]
This should be resolved in v7 with #14505
